### DISCUSSION TITLE
kompot.batch_call api

### DISCUSSION THREAD

**Discussion by hugetim** [anvilistas/anvil-labs#69] [Ideas]
Now I'm wondering whether it might be better to use `results` as [the name of the attribute collecting the return values](https://github.com/anvilistas/anvil-labs/pull/68/files#r858823103).

Also, it's now striking me as odd to reference the context manager object (`c` in the [example code](https://anvil-labs.readthedocs.io/en/latest/guides/modules/kompot.html#batch_call)) outside of the `with` statement context. I think I've just never seen a use of a `with` statement like that. Does that seem fine to others?

**Comment by meatballs** [comment]
> I think I've just never seen a use of a with statement like that. Does that seem fine to others?

Perfectly normal. You'll find plenty of example using files with something like:
```python
with open("some_file", "w") as f:
    f.write("hello world")
```

**Comment by hugetim** [comment]
Right, but you don't usually see `f` referenced after the indented context, like:

```python
with open("some_file", "w") as f:
    f.write("hello world")
assert(f.closed)
```

But I guess there's no problem with it.

**Reply by s-cork** [reply]
Checkout the context manager in unittest for assertRaises. 

```python

with self.assertRaises(ZeroDivisionError) as e:
  1/0
self.assertIn("zero", repr(e.exception))

```

So also perfectly normal. 

Buyers remorse?

result/results.
It's the result of doing all the server calls. But it doesn't much bother me.

**Reply by hugetim** [reply]
I just thought the plural would help signal that it returns a list containing the results from the multiple server functions called. But no biggie.
